src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -956,30 +988,25 @@ InstanceKlass* SystemDictionaryShared::prepare_shared_lambda_proxy_class(Instanc
 
 void SystemDictionaryShared::check_verification_constraints(InstanceKlass* klass,
                                                             TRAPS) {
-  assert(!CDSConfig::is_dumping_static_archive() && CDSConfig::is_using_archive(), "called at run time with CDS enabled only");
+  assert(CDSConfig::is_using_archive(), "called at run time with CDS enabled only");
   RunTimeClassInfo* record = RunTimeClassInfo::get_for(klass);
 
   int length = record->num_verifier_constraints();
-  if (length > 0) {
+  if (length > 0 || klass->name()->equals("HelloWorld")) {
```

Comment:
Is the "HelloWorld" check leftover from debugging?

Reply:
Fixed.